core/src/main/java/org/polypheny/db/algebra/core/lpg/LpgMatch.java

```diff
@@ -70,10 +70,10 @@ protected AlgDataType deriveRowType() {
 
     @Override
     public String algCompareString() {
-        return "$" + getClass().getSimpleName() +
-                "$" + matches.hashCode() +
-                "$" + names.hashCode() +
-                "$" + input.algCompareString();
+        return getClass().getSimpleName() + "$" +
+                matches.hashCode() + "$" +
+                names.hashCode() + "$" +
```

Comment:
With documents, the names have been strings. In LPG, they are PolyString. Is there a reason? Further, is the hash code stable enough for the purposes of this compare sting?